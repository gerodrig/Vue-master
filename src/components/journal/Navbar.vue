
<template>
    <nav class="flex items-center justify-between h-16 bg-green-900">
        <a class="text-2xl text-white">
            <img src="@/assets/vue.svg" alt="Vue Logo" class="inline-block w-10 h-10 mx-2">
            Daybook
        </a>

        <div @click="onLogout" class="right-0 p-4 mr-2 text-white hover:cursor-pointer">
            <span class="mr-2">{{ username }}</span>
            <button>
                <i class="text-2xl hover:text-yellow-500 fa fa-sign-out-alt"></i>
            </button>
        </div>
    </nav>
</template>

<script lang='ts'>
import { useRouter } from 'vue-router';
 import useAuth from '@/composables/auth/useAuth';
    export default {
        name: 'Navbar',
        setup() {
            const router = useRouter();
            const { username, logout } = useAuth();

            return {
                username,
                
                //methods
                onLogout: () => {
                    router.push({ name: 'login' })
                    logout();
                }
            }
        }
    };
</script>
