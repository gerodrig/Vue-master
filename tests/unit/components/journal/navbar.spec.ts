import { describe, expect, it, vi, Mock, beforeEach} from 'vitest';
import { shallowMount } from '@vue/test-utils';
import { useRouter } from 'vue-router';

import Navbar from '@/components/journal/Navbar.vue';
import createVuexStore from 'tests/unit/mocks/store/mock-store';

// const mockStore = {
//     getters: {
//         'auth/currentAuthState': 'authenticated',
//         'auth/currentUsername': 'test',
//     }
// }

// vi.mock('vuex', () => ({
//     useStore: () => mockStore
// }));
const store = createVuexStore({
  status: 'authenticated',
  user: {
    username: 'Benito',
    email: 'benito@test.com',
  },
  idToken: '1234567890',
  refreshToken: '0987654321',
});


vi.mock('vue-router', () => ({
  useRouter: vi.fn(() => ({
    push: vi.fn(),
  })),
}));

describe('Testing Navbar component functionality for journal', () => {

    (useRouter as Mock).mockReturnValue({
        push: vi.fn(),
      })

    beforeEach(() => {
        (useRouter().push as Mock).mockReset();
        vi.clearAllMocks();
    });

  it('should renders the Navbar component successfully', () => {
    const wrapper = shallowMount(Navbar, {
        global: {
            plugins: [store],
        }
    });

    const span = wrapper.find('div').find('span');

    expect(wrapper.html()).toMatchSnapshot();
    expect(span.text()).toBe('Benito');
  });

  it('should test that logout button works', async () => {
    const wrapper = shallowMount(Navbar, {
        global: {
            plugins: [store],
        }
    });

    await wrapper.find('div').trigger('click');

    expect( store.state.auth ).toEqual({
        status: 'not-authenticated',
        user: null,
        idToken: null,
        refreshToken: null,
    });

    expect(useRouter().push).toHaveBeenCalledWith({ name: 'login' });

  });
});
