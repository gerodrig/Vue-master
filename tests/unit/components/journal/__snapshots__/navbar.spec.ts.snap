// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing Navbar component functionality for journal > should renders the Navbar component successfully 1`] = `
"<nav class=\\"flex items-center justify-between h-16 bg-green-900\\"><a class=\\"text-2xl text-white\\"><img src=\\"/src/assets/vue.svg\\" alt=\\"Vue Logo\\" class=\\"inline-block w-10 h-10 mx-2\\"> Daybook </a>
  <div class=\\"right-0 p-4 mr-2 text-white hover:cursor-pointer\\"><span class=\\"mr-2\\">Benito</span><button><i class=\\"text-2xl hover:text-yellow-500 fa fa-sign-out-alt\\"></i></button></div>
</nav>"
`;
